sdk/typescript/src/builder/TransactionBlock.ts

```diff
@@ -277,6 +289,18 @@ export class TransactionBlock {
 		return inserted ?? this.#input('object', value);
 	}
 
+	/**
+	 * Add a new object input to the transaction.
+	 */
+	receiving(value: ObjectId | ObjectCallArg) {
+		const id = getIdFromCallArg(value);
+		// deduplicate
+		const inserted = this.#blockData.inputs.find(
+			(i) => i.type === 'object' && id === getIdFromCallArg(i.value),
+		);
+		return inserted ?? this.#input('object', value);
```

Comment:
Should this also be an `object` input? It looks the same as the `object` helper, above.  I noticed below that both `object` and `receiving` result in a type of `object` and then a different `objectType`, so maybe this works fine, but what happens if you pass a shared object arg into `receiving`?  Seems like something we should protect against.

Reply:
It should work out since the actual building of the argument is done after-the-fact here https://github.com/MystenLabs/sui/blob/main/sdk/typescript/src/builder/TransactionBlock.ts#L668C1-L668C1 and that does checking at the time these placeholders are "inflated" to actual `ReceivingRef`/`ObjectRef`/`SharedObjectRef` arguments. 

I kept `Receiving` as `object` since that lines up with what we have elsewhere in the codebase both in TS and in Rust for receiving (it's an `ObjectArg` with a `Receiving` variant in TS and in Rust). 

It's a good idea to add a test or two for this though so will add that in :)

Reply:
Added a test for trying to pass a shared object to a `receiving` argument.